include/seqan3/argument_parser/detail/format_ctd.hpp

```diff
@@ -0,0 +1,484 @@
+     *
+     * \details
+     * For the sake of the CTD exporter, add_flag is the same of add_option, with a constrained option_type and
+     * a default validator.
+     */
+    void add_flag(bool & value,
+                  char const short_id,
+                  std::string const & long_id,
+                  std::string const & desc,
+                  option_spec const & spec) 
+    {
+        add_option(value, 
+                   short_id,
+                   long_id,
+                   desc,
+                   spec,
+                   default_validator<bool> {});
+    }
+
+    /*!\brief Adds calls for appending 'clielement' and 'ITEM' nodes to the CTD DOM tree to be evaluated at parse time.
+     *
+     * \tparam option_type    The type of variable in which to store the given command line argument.
+     * \tparam validator_type The type of validator applied to the value after parsing.
+     *
+     * \param[out] value     The variable in which to store the given command line argument.
+     * \param[in]  desc      The description of the positional option.
+     * \param[in]  validator The validator applied to the value after parsing (callable).
+     */
+    template<typename option_type, typename validator_type>
+    void add_positional_option(option_type & value,
+                               std::string const & desc,
+                               validator_type && validator) 
+    {
+        unsigned argument_id = args_counter;
+
+        clielement_argument_callbacks.push_back([this,
+						 argument_id] (std::string app_name) {
+            append_ctd_clielement_node("",
+                                       build_reference_name(app_name,
+                                                            argument_id).data());
+        });
+
+        // Register ITEM callback.
+        item_argument_callbacks.push_back([this,
+                                           value,
+                                           desc,
+                                           validator,
+					   argument_id] () {
+            append_ctd_item_node(build_argument_name(argument_id),
+                                 get_type_as_gkn_string(value,
+                                                        validator),
+                                 desc,
+                                 "",
+                                 "*.*",
+                                 "true",
+                                 "false",
+                                 "");
+        });
+
+        args_counter++;
+    }
+
+    /*!\brief Builds the CTD document DOM tree and prints it to standard output.
+     *
+     * \param[in] meta The meta information that are needed for building the DOM tree.
+     */
+    void parse(argument_parser_meta_data const & meta) 
+    {
+        // Validate the application name using the regex provided by the CTD XML schema.
+        if (!std::regex_match(meta.app_name, 
```

Comment:
ok interesting. I have to check, because we recently added requirements on the app name on a level above. Seems to be the same.

Reply:
I will remove the check from the CTD formatter, as the check implemented in the argument_parser class is even more stringent